plugins/mission/mission_impl.cpp

```diff
@@ -932,4 +934,135 @@ void MissionImpl::process_timeout()
+    }
+
+    // Compose mission items
+    return compose_mission_items(mission_items, parsed_plan);
+}
+
+Mission::Result MissionImpl::compose_mission_items(MAV_CMD cmd, std::vector<double> params,
+                                                   std::shared_ptr<MissionItem> &new_mission_item,
+                                                   Mission::mission_items_t &mission_items)
+{
+    Mission::Result result = Mission::Result::SUCCESS;
+
+    // Choosen "Do-While(0)" loop for the convenience of using `break` statement.
+    do {
+        if (cmd == MAV_CMD_NAV_WAYPOINT ||
+            cmd == MAV_CMD_NAV_TAKEOFF ||
+            cmd == MAV_CMD_NAV_LAND) {
+            if (new_mission_item->has_position_set()) {
+                mission_items.push_back(new_mission_item);
+                new_mission_item = std::make_shared<MissionItem>();
+            }
+
+            if (cmd == MAV_CMD_NAV_WAYPOINT) {
+                auto is_fly_thru = !(int(params[0]) > 0);
+                new_mission_item->set_fly_through(is_fly_thru);
+            }
+            auto lat = params[4], lon = params[5];
+            new_mission_item->set_position(lat, lon);
+
+            auto rel_alt = float(params[6]);
+            new_mission_item->set_relative_altitude(rel_alt);
+
+        } else if (cmd == MAV_CMD_DO_MOUNT_CONTROL) {
+            auto pitch = float(params[0]), yaw = float(params[2]);
+            new_mission_item->set_gimbal_pitch_and_yaw(pitch, yaw);
+
+        } else if (cmd == MAV_CMD_NAV_LOITER_TIME) {
+            auto loiter_time_s = float(params[0]);
+            new_mission_item->set_loiter_time(loiter_time_s);
+
+        } else if (cmd == MAV_CMD_IMAGE_START_CAPTURE) {
+            auto photo_interval = int(params[1]),  photo_count = int(params[2]);
+
+            if (photo_interval > 0 && photo_count == 0) {
+                new_mission_item->set_camera_action(MissionItem::CameraAction::START_PHOTO_INTERVAL);
+                new_mission_item->set_camera_photo_interval(photo_interval);
+            } else if (photo_interval == 0 && photo_count == 1) {
+                new_mission_item->set_camera_action(MissionItem::CameraAction::TAKE_PHOTO);
+            } else {
+                LogErr() << "Mission item START_CAPTURE params unsupported.";
+                result = Mission::Result::UNSUPPORTED;
+                break;
+            }
+
+        } else if (cmd == MAV_CMD_IMAGE_STOP_CAPTURE) {
+            new_mission_item->set_camera_action(MissionItem::CameraAction::STOP_PHOTO_INTERVAL);
+
+        } else if (cmd == MAV_CMD_VIDEO_START_CAPTURE) {
+            new_mission_item->set_camera_action(MissionItem::CameraAction::START_VIDEO);
+
+        } else if (cmd == MAV_CMD_VIDEO_STOP_CAPTURE) {
+            new_mission_item->set_camera_action(MissionItem::CameraAction::STOP_VIDEO);
+
+        } else if (cmd == MAV_CMD_DO_CHANGE_SPEED) {
+            enum { AirSpeed, GroundSpeed };
```

Comment:
You are using `AirSpeed` and `GroundSpeed` as `int` but don't set the int values which seems dangerous if a compiler does not use the usual 0, 1, 2...

Reply:
Oh! Ok, Did you mean to do this: `enum { AirSpeed = 0, GroundSpeed = 1 };`
Is that possible ? I am not sure.

Reply:
Yes, that's certainly possible.

Reply:
OK! Thanks.